Tools/autotest/arducopter.py

```diff
@@ -902,6 +902,61 @@ def ensure_smartrtl(mav, m):
         self.set_parameter('FS_OPTIONS', 0)
         self.progress("All GCS failsafe tests complete")
 
+    def test_custom_controller(self, timeout=300):
+        self.progress("Configure custom controller paramters")
+        self.set_parameters({
+            'CC_TYPE': 2,
+            'CC_AXIS_MASK': 7,
+            'RC6_OPTION': 109,
+        })
+        self.set_rc(6, 1000)
+        self.reboot_sitl()
+
+        cc_type = self.get_parameter("CC_TYPE")
+
+        if cc_type != 2 :
+            NotAchievedException("Custom controller is not switched to PID backend.")
+
+        # check if we can retrive any param inside PID backend
+        self.get_parameter("CC2_RAT_YAW_P")
+
+        # takeoff in GPS mode and switch to CIRCLE
+        self.takeoff(10, mode="LOITER", takeoff_throttle=2000)
+        self.change_mode("CIRCLE")
+
+        # switch custom controller on
+        self.wait_statustext(
+            text="Custom controller is ON",
+            timeout=30,
+            the_function=lambda: self.set_rc(6, 2000))
```

Comment:
```suggestion
        self.context_push()

        # switch custom controller on
        self.set_rc(6, 2000)
        self.wait_statustext("Custom controller is ON")
```
I'm assuming you *really* don't need 30 seconds to turn the controller on :-)

.... then pop the context off right down the bottom of the test.

Reply:
I pushed the changes. Can you check again?